src/transport_interface/transport_interface_test.c

```diff
@@ -421,26 +522,137 @@ static void prvSendRecvCompareFunc( void * pParam )
     for( testBufferSize = 1; testBufferSize <= TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH; testBufferSize = testBufferSize + 1 )
     {
         /* Initialize the test data buffer. */
-        prvInitializeTestData( pTrasnportTestBufferStart, testBufferSize );
+        prvInitializeTestData( pTransportTestBufferStart, testBufferSize );
 
         /* Send the test data to the server. */
         if( pThreadParameter->xResult == true )
         {
-            pThreadParameter->xResult = prvTransportSendData( pTestTransport, pNetworkContext, pTrasnportTestBufferStart,
+            pThreadParameter->xResult = prvTransportSendData( pTestTransport,
+                                                              pNetworkContext,
+                                                              pTransportTestBufferStart,
+                                                              testBufferSize );
+        }
+
+        /* Receive the test data from server. */
+        if( pThreadParameter->xResult == true )
+        {
+            pThreadParameter->xResult = prvTransportRecvData( pTestTransport,
+                                                              pNetworkContext,
+                                                              pTransportTestBufferStart,
                                                               testBufferSize );
         }
 
+        /* Compare the test data received from server. */
+        if( pThreadParameter->xResult == true )
+        {
+            pThreadParameter->xResult = prvVerifyTestData( pTransportTestBufferStart,
+                                                           testBufferSize,
+                                                           TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH );
+        }
+
+        if( ( pThreadParameter->stopFlag == true ) || ( pThreadParameter->xResult == false ) )
+        {
+            break;
+        }
+
+        #if ( TRANSPORT_TEST_PRINT_DEBUG_PROGRESS == 1 )
+            /* Output information to indicate the test is running. */
+            UNITY_OUTPUT_CHAR( '.' );
+        #endif
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+#ifdef TRANSPORT_TEST_EXECUTE_WRITEV_TESTS

```

Comment:
More of a question, but is there a reason that there are multiple IFDEFs for this? Would it make sense to move this function, and other ones related to the TRANSPORT_TEST_EXECUTE_WRITEV_TESTS macro, up to the first #ifdef/#endif?
My thought process being that way all the code relevant to these tests is in a single block so if a user does a grep for the macro it only shows up once.